docs/manual/working/scalaGuide/main/json/code/ScalaJsonAutomatedSpec.scala

```diff
@@ -320,12 +276,15 @@ class ScalaJsonAutomatedSpec extends Specification {
       //#auto-JSON-custom-trait
       import play.api.libs.json._
 
-      implicit val cfg = JsonConfiguration(
+      implicit val cfg: JsonConfiguration = JsonConfiguration(
         // Each JSON objects is marked with the admTpe, ...
         discriminator = "admTpe",
         // ... indicating the lower-cased name of sub-type
-        typeNaming = JsonNaming { fullName =>
```

Comment:
Needed change because of a limitation in Dotty's typeclass derivation: lampepfl/dotty#11048.  You can also see a reference to that ticket in this diff.

Reply:
Meaning there is a regression for the existing Scala2 play-json?

Reply:
No, I kept the behaviour the same for Scala 2.  That's why there is both the original fully qualified names, mapped to lowercase, and the new simple class names, which is all Scala 3 has access to.

Reply:
would be possible to use a macro to get the "FQCN" (I know not all types are classes) so this is not a breaking change when people migrate to dotty?

Reply:
Yeah, I assume that would be possible using an accompanying macro.  I'm thinking at the start of the Scala 3 inline def you could try to summon the fully qualified names of the subclasses, and zip that with the element types (instead of using the mirror labels).

I'll wait to see if this is requested by downstream before spending more time on this.

Reply:
This will be a problem for my work projects for sure...